### PR TITLE
Add ApplicationInsights client metrics capture

### PR DESCRIPTION
**Pull request by tnavarra**
I verified this against an AppInsights resource that I have and it is logging metrics.  I followed the documentation somewhat from [here](https://docs.microsoft.com/en-us/azure/azure-monitor/app/javascript) and [here](https://github.com/microsoft/ApplicationInsights-JS/blob/17ef50442f73fd02a758fbd74134933d92607ecf/extensions/applicationinsights-react-js/README.md).  However, the React plugin for app insights specifically talks about metrics for class based components.  Nowhere does it discuss components using hooks.  I'm still very green with React so perhaps someone has a different interpretation of these docs.

### CODE REVIEW

**Comment by jjh121** [comment]
> I verified this against an AppInsights resource that I have and it is logging metrics. I followed the documentation somewhat from [here](https://docs.microsoft.com/en-us/azure/azure-monitor/app/javascript) and [here](https://github.com/microsoft/ApplicationInsights-JS/blob/17ef50442f73fd02a758fbd74134933d92607ecf/extensions/applicationinsights-react-js/README.md). However, the React plugin for app insights specifically talks about metrics for class based components. Nowhere does it discuss components using hooks. I'm still very green with React so perhaps someone has a different interpretation of these docs.

I don't think that the react appinsights plugin is going to work for us with hooks. I haven't tried it, but I did find [this](https://github.com/microsoft/ApplicationInsights-JS/issues/991) and [this guide on how to make it work](https://www.aaron-powell.com/posts/2019-11-19-combining-react-hooks-with-appinsights/)

File: FoxHound.Client/src/.env [hunk 1]
```diff
@@ -1 +1,2 @@
-API_URL=https://localhost:44360/api
\ No newline at end of file
+API_URL=https://localhost:44360/api
+APP_INSIGHTS_INSTRUMENTATION_KEY=
```
Comment (jjh121) [inline FoxHound.Client/src/.env:2]: Does leaving the instrumentation key null cause any issues when it's not found (i.e. running locally?

Comment (tnavarra) [reply FoxHound.Client/src/.env:2]: No issues when running locally without it.  It simply will not push App Insights metrics anywhere.

**Comment by tnavarra** [comment]
> > I verified this against an AppInsights resource that I have and it is logging metrics. I followed the documentation somewhat from [here](https://docs.microsoft.com/en-us/azure/azure-monitor/app/javascript) and [here](https://github.com/microsoft/ApplicationInsights-JS/blob/17ef50442f73fd02a758fbd74134933d92607ecf/extensions/applicationinsights-react-js/README.md). However, the React plugin for app insights specifically talks about metrics for class based components. Nowhere does it discuss components using hooks. I'm still very green with React so perhaps someone has a different interpretation of these docs.
> 
> I don't think that the react appinsights plugin is going to work for us with hooks. I haven't tried it, but I did find [this](https://github.com/microsoft/ApplicationInsights-JS/issues/991) and [this guide on how to make it work](https://www.aaron-powell.com/posts/2019-11-19-combining-react-hooks-with-appinsights/) ... is this working because you converted the App.tsx component into a class based component?
> 
> I actually think trying to make the custom hook(s) outlined in that guide would be pretty interesting. It would allow us to send custom messages in code to app insights if i'm understanding it correctly.

I read over the linked blog posts.  If you take a look at the first blog post in the series you'll see a very similar wire up of App Insights as to what's in this PR.  I think the main difference to note is that he offloads that to its own file.  I think that makes a lot of sense and I've modified this PR to do the same.  From what I can tell there is nothing special in the configuration to get App Insights to work with function based components.  The rest of the setup with the App Insights context and custom hook appears to be purely for tracing custom events using a hook.  It probably does make sense to add this but perhaps that should be separate PR once we know thee basic setup is working.  Thoughts?

**Comment by jjh121** [comment]
> I read over the linked blog posts. If you take a look at the first blog post in the series you'll see a very similar wire up of App Insights as to what's in this PR. I think the main difference to note is that he offloads that to its own file. I think that makes a lot of sense and I've modified this PR to do the same. From what I can tell there is nothing special in the configuration to get App Insights to work with function based components. The rest of the setup with the App Insights context and custom hook appears to be purely for tracing custom events using a hook. It probably does make sense to add this but perhaps that should be separate PR once we know thee basic setup is working. Thoughts?

I think this makes a lot of sense. I like the separate file for AppInsights. I do think it'd be interesting to implement the custom tracing hook and another PR makes a lot of sense.

**Pull request merged**